### PR TITLE
Meenu/Live market price

### PR DESCRIPTION
**Pull request by meenakshi-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/home/live-pricing/_utils.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { TMarketButtons } from './_types'
+
+export const TABLE_VISIBLE_ROWS = 6
+
+export const market_buttons: TMarketButtons[] = [
+    {
+        id: 0,
+        button_text: 'Forex',
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/_utils.ts:8]: These title should be translatable

File: src/pages/home/live-pricing/components/_elements.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+const Container = styled(Cell)`
+    justify-content: center;
+`
+
+const DailyPercentageText = styled(HeaderText)<{ bull?: boolean }>`
+    ${({ bull }) =>
+        bull
+            ? css`
+                  color: #4bb4b3;
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/components/_elements.tsx:138]: let's use variable for this color

File: src/pages/home/live-pricing/_use-live-columns.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { TableCell, TableHeaderCell, DailyPercentageCell } from './components/_elements'
+import { TMarketData } from './_types'
+
+const liveMarketColumnHelper = createColumnHelper<TMarketData>()
+
+const useLiveColumns = (marketData: () => void) => {
+    const columns = useMemo(() => {
+        return [
+            liveMarketColumnHelper.accessor('symbol', {
+                header: () => <TableHeaderCell text={'Trading Pairs'} />,
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/_use-live-columns.tsx:12]: Content should be translatable

File: src/pages/home/live-pricing/components/_elements.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    }
+    to {
+        transform: rotate(360deg);
+    }
+`
+
+export const Spinner = styled.div`
+    animation: ${rotate360} 1s linear infinite;
+    transform: translateZ(0);
+    border-top: 2px solid grey;
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/components/_elements.tsx:35]: use variables for the colors

File: src/pages/home/live-pricing/index.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                            }}
+                        >
+                            <Header type="paragraph-2">{marketItem.button_text}</Header>
+                        </MarketButton>
+                    ))}
+                </MarketsContainer>
+            </Markets>
+            <ContainerWrapper>
+                <Header type="paragraph-1" weight="normal" align="center">
+                    Benefit from round-the-clock trading hours (Monday to Friday), high liquidity,
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/index.tsx:118]: Same here

File: src/pages/home/live-pricing/index.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                    low barriers to entry, a wide range of offerings, and opportunities to trade on
+                    world events.
+                </Header>
+                <LiveMarketTable market={selected_market} />
+                <DisclaimerText>
+                    Disclaimer: All spreads are indicative. To view real-time spreads, clients
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/index.tsx:124]: Same here

File: src/pages/home/live-pricing/components/_live_market_table.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        <TableContainer>
+            {is_loading && (
+                <TableLoadingContainer>
+                    <Spinner />
+                </TableLoadingContainer>
+            )}
+            <Table>
+                <thead>
+                    {table.getHeaderGroups().map((headerGroup) => (
+                        <TableRow key={headerGroup.id} bg="#f9f9f9">
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/components/_live_market_table.tsx:87]: Add the background color from css

File: src/pages/home/live-pricing/_use-live-columns.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import React, { useMemo } from 'react'
+import { TableCell, TableHeaderCell, DailyPercentageCell } from './components/_elements'
+import { TMarketData } from './_types'
+
+const liveMarketColumnHelper = createColumnHelper<TMarketData>()
+
+const useLiveColumns = (marketData: () => void) => {
+    const columns = useMemo(() => {
+        return [
+            liveMarketColumnHelper.accessor('symbol', {
```
Comment (habib-deriv) [inline src/pages/home/live-pricing/_use-live-columns.tsx:11]: Create an object of the static content and generate this array of items using "for...in" loop in by a function

Comment (habib-deriv) [reply src/pages/home/live-pricing/_use-live-columns.tsx:11]: It's needed to get rid of the duplication of the same array items

File: src/pages/home/live-pricing/components/_live_market_table.tsx [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        setIsLoading(true)
+
+        send(
+            { trading_platform_asset_listing: 1, platform: 'mt5', type: 'brief' },
+            (response: TMarketDataResponse) => {
+                const responseData = [...response.trading_platform_asset_listing.mt5.assets]
+                const markets = new Map<TAvailableLiveMarkets, TMarketData[]>()
+
+                responseData.forEach((item) => {
+                    const currentMarket = [...(markets.get(item.market) ?? [])]
```
Comment (NikitK-deriv) [inline src/pages/home/live-pricing/components/_live_market_table.tsx:47]: Seems it always should return empty due the markets is empty from beginning
BTW for workable function we don't need `?? []`

File: src/pages/home/live-pricing/index.tsx [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                        <>
+                            <MarketButton
+                                selected={marketItem.market_name === selected_market}
+                                key={marketItem.id}
+                                onClick={() => {
+                                    onMarketButtonClick(marketItem.market_name)
+                                }}
+                            >
+                                <Header type="paragraph-2">
+                                    {<Localize translate_text={marketItem.button_text} />}
```
Comment (NikitK-deriv) [inline src/pages/home/live-pricing/index.tsx:118]:
```suggestion
                                    <Localize translate_text={marketItem.button_text} />
```

File: src/components/hooks/use-deriv-api.tsx [hunk 11]
```diff
@@ -28,8 +28,16 @@ export const useDerivApi = () => {
     const send = useCallback(async (data: object, callback: (e: object) => void) => {
         if (ws) {
             const response = await ws.current.send(data)
-
-            callback(response)
+            callback?.(response)
```
[the remark this answers is not in the document]
Comment (meenakshi-deriv) [reply src/components/hooks/use-deriv-api.tsx]: hi @NikitK-deriv Thanks for the comments. It's all resolved in new commit

**Review by habib-deriv — DISMISSED** [review]

**Review by habib-deriv — APPROVED** [review]

**Pull request closed without merging**